+++ /usr/share/X11/xkb/symbols/cz	2019-01-08 19:16:47.148009758 +0100
@@ -78,4 +78,34 @@
     key <BKSL>  { [ backslash,        bar,        slash,     NoSymbol ] };
 };
 
+partial alphanumeric_keys alternate_group
+xkb_symbols "cz_programmer" {
+
+    // us layout with czech chars under AltGr
+
+    include "us"
+    name[Group1]= "Czech programmer's";
+
+    key <TLDE>{[     grave,  asciitilde,     semicolon, dead_abovering ]};
+    key <AE02>{[         2,          at,        ecaron,     Ecaron ]};
+    key <AE03>{[         3,  numbersign,        scaron,     Scaron ]};
+    key <AE04>{[         4,      dollar,        ccaron,     Ccaron ]};
+    key <AE05>{[         5,     percent,        rcaron,     Rcaron ]};
+    key <AE06>{[         6, asciicircum,        zcaron,     Zcaron ]};
+    key <AE07>{[         7,   ampersand,        yacute,     Yacute ]};
+    key <AE08>{[         8,    asterisk,        aacute,     Aacute ]};
+    key <AE09>{[         9,   parenleft,        iacute,     Iacute ]};
+    key <AE10>{[         0,  parenright,        eacute,     Eacute ]};
+    key <AE12>{[     equal,        plus,    dead_acute, dead_caron ]};
+
+    key <AD03>{[         e,           E,             e,   EuroSign ]};
+    key <AD11>{[ bracketleft, braceleft,        uacute,     Uacute ]};
+
+    key <AC10>{[ semicolon,      colon,          uring,      Uring ]};
+    key <AC11>{[ apostrophe,  quotedbl,        section,   quotedbl ]};
+    key <BKSL>{[ backslash,        bar, dead_diaeresis, apostrophe ]};
+
+    include "level3(ralt_switch)"
+};
+
 partial alphanumeric_keys
+++ /usr/share/X11/xkb/rules/base.xml	2019-01-08 19:35:11.003050528 +0100
@@ -2769,6 +2769,12 @@
             <description>Czech (with &lt;\|&gt; key)</description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>cz_programmer</name>
+            <description>Czech programmer's</description>
+          </configItem>
+        </variant>
         <variant>
           <configItem>
             <name>qwerty</name>
+++ /usr/share/X11/xkb/rules/base.lst	2019-01-08 18:56:18.505579275 +0100
@@ -418,2 +418,3 @@
   bksl            cz: Czech (with &lt;\|&gt; key)
+  cz_programmer   cz: Czech (Czech programmer's)
   qwerty          cz: Czech (QWERTY)
